validator/testcases/javascript/predefinedentities.py

```diff
@@ -121,8 +216,18 @@
          "value": {}},
     u"nsIBrowserSearchService":
         {"value":
-            {u"currentEngine": {"readonly": True},
-             u"defaultEngine": {"readonly": True}}},
+            {u"currentEngine":
+                {"readonly": search_warning(severity="high")},
```

Comment:
`"dangerous"` applies to any access, or in the case of functions, to calls. `"readonly"` only applies to attempts to write to the value, which is what's important for these cases.

As for `True` or `False`, `"readonly"` has accepted `True`/`False`/`lambda` for a long time. This patch changes it to also accept a message description or a `dict` of message properties the same way `"dangerous"` does.